velox/vector/LazyVector.cpp

```diff
@@ -222,6 +214,10 @@ void LazyVector::ensureLoadedRows(
     SelectivityVector& baseRows) {
   decoded.decode(*vector, rows, false);
   ensureLoadedRowsImpl(vector, decoded, rows, baseRows);
+  // An explicit call to loadedVector() is necessary to allow for proper
+  // initialization of dictionaries, sequences, etc. on top of lazy vector
+  // after it has been loaded, even if loaded via some other path.
+  vector->loadedVector();
```

Comment:
if this pattern is nested, like `dict(row(lazy(dict(row(lazy)))))`, will the inner dictionary being initialized properly?  Can we add a test for that?

Reply:
good point, let me check, if this recursively does not happen then we might have to overhaul these mechanisms to support such cases